File: python/majority-element.py
from typing import List


class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        val = nums[0]
        count = 1
        for num in nums:
            if num != val:
                count -= 1
                if count == 0:
                    val = num
                    count = 1
            else:
                count += 1
        return val

File: rust/majority-element.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut num = nums[0];
        let mut count = 1;
        for i in 0..nums.len() {
            if nums[i] != num {
                count -= 1;
                if count == 0 {
                    num = nums[i];
                    count = 1;
                }
            } else {
                count += 1;
            }
        }
        return num;
    }
}